crates/bevy_utils/src/short_names.rs

```diff
@@ -59,7 +59,20 @@ pub fn get_short_name(full_name: &str) -> String {
 
 #[inline(always)]
 fn collapse_type_name(string: &str) -> &str {
-    string.split("::").last().unwrap()
+    // Enums types are retained.
+    // As heuristic, we assume the enum type to be uppercase.
+    let mut segments = string.rsplit("::");
+    let (last, second_last): (&str, Option<&str>) = (segments.next().unwrap(), segments.next());
+    let Some(second_last) = second_last else {
+        return last;
+    };
+
+    if second_last.starts_with(|c: char| c.is_uppercase()) {
```

Comment:
This is more in line with common rust usage.

```suggestion
    if second_last.starts_with(char::is_uppercase) {
```